packages/ui-date-input/src/DateInput2/README.md

```diff
@@ -33,51 +42,149 @@ This component is an updated version of [`DateInput`](/#DateInput) that's easier
+            }}
+            value={inputValue}
+            width="20rem"
+            onChange={(e, inputValue, dateString) => {
+              setInputValue(inputValue)
+              setDateString(dateString)
+            }}
+            invalidDateErrorMessage="Invalid date"
+          />
+          <p>
+            Input Value: <code>{inputValue}</code>
+            <br />
+            UTC Date String: <code>{dateString}</code>
+          </p>
+        </div>
+    )
+  }
+
+  render(<Example />)
+  ```
+
+### Parsing and formatting dates
+
+When typing in a date manually (instead of using the included picker), the component tries to parse the date as you type it in. By default parsing is based on the user's locale which determines the order of day, month and year (e.g.: a user with US locale will have MONTH/DAY/YEAR order, and someone with GB locale will have DAY/MONTH/YEAR order).
+
+Any of the following separators can be used when typing a date: `,`, `-`, `.`, `/` or a whitespace however on blur the date will be formatted according to the locale and separators will be changed and leading zeros also adjusted.
+
+If you want different parsing and formatting then the current locale you can use the `dateFormat` prop which accepts either a string with a name of a different locale (so you can use US date format even if the user is France) or a parser and formatter functions.
+
+The default parser also have a limitation of excluding years before `1000` and after `9999`. These values are invalid by default but not with custom parsers.
+
+```js
+---
+type: example
+---
+const Example = () => {
+  const [value, setValue] = useState('')
+  const [value2, setValue2] = useState('')
+
+  return (
+    <div>
+      <p>US locale with german date format:</p>
       <DateInput2
         renderLabel="Choose a date"
         screenReaderLabels={{
           calendarIcon: 'Calendar',
           nextMonthButton: 'Next month',
           prevMonthButton: 'Previous month'
         }}
-        value={value}
         width="20rem"
+        value={value}
+        locale="en-us"
+        dateFormat="de-de"
         onChange={(e, value) => setValue(value)}
-        invalidDateErrorMessage="Invalid date"
       />
-    )
-  }
+      <p>US locale with ISO date format:</p>
+      <DateInput2
+        renderLabel="Choose a date"
+        screenReaderLabels={{
+          calendarIcon: 'Calendar',
+          nextMonthButton: 'Next month',
+          prevMonthButton: 'Previous month'
+        }}
+        width="20rem"
+        value={value2}
+        messages={[
+          {type: 'hint', text: 'YYYY-MM-DD'}
+        ]}
+        locale="en-us"
+        dateFormat={{
+          parser: (input) => {
+            // split input on '.', whitespace, '/', ',' or '-' using regex: /[.\s/.-]+/
+            // the '+' allows splitting on consecutive delimiters
+            const [year, month, day] = input.split(/[,.\s/.-]+/)
+            const newDate = new Date(year, month-1, day)
```

Comment:
This is interpreted in local time. I think it could lead to bugs if the component uses a different timezone -- it might return a date a day before/after?

Reply:
might be true, I will check that